python/dgl/data/lrgb.py

```diff
@@ -0,0 +1,237 @@
+import hashlib
+import os
+import pickle
+
+import pandas as pd
+from ogb.utils import smiles2graph
+from tqdm import tqdm
+
+from .. import backend as F
+
+from ..convert import graph as dgl_graph
+from .dgl_dataset import DGLDataset
+from .utils import download, load_graphs, save_graphs, Subset
+
+
+class PeptidesStructuralDataset(DGLDataset):
+    r"""Peptides structure dataset for the graph classification task.
+
+    DGL dataset of 15,535 small peptides represented as their molecular
+    graph (SMILES) with 11 regression targets derived from the peptide's
+    3D structure.
+
+    The original amino acid sequence representation is provided in
+    'peptide_seq' and the distance between atoms in 'self_dist_matrix' field
+    of the dataset file, but not used here as any part of the input.
+
+    The 11 regression targets were precomputed from molecule XYZ:
+        Inertia_mass_[a-c]: The principal component of the inertia of the
+            mass, with some normalizations. Sorted
+        Inertia_valence_[a-c]: The principal component of the inertia of the
+            Hydrogen atoms. This is basically a measure of the 3D
+            distribution of hydrogens. Sorted
+        length_[a-c]: The length around the 3 main geometric axis of
+            the 3D objects (without considering atom types). Sorted
+        Spherocity: SpherocityIndex descriptor computed by
+            rdkit.Chem.rdMolDescriptors.CalcSpherocityIndex
+        Plane_best_fit: Plane of best fit (PBF) descriptor computed by
+            rdkit.Chem.rdMolDescriptors.CalcPBF
+
+    Statistics:
+    Train examples: 10,873
+    Valid examples: 2,331
+    Test examples: 2,331
+    Average number of nodes: 150.94
+    Average number of edges: 307.30
+    Number of atom types: 9
+    Number of bond types: 3
+
+    Parameters
+    ----------
+    raw_dir : str
+        Raw file directory to download/contains the input data directory.
+        Default: "~/.dgl/".
+    force_reload : bool
+        Whether to reload the dataset.
+        Default: False.
+    verbose : bool
+        Whether to print out progress information.
+        Default: False.
+    smiles2graph (callable):
```

Comment:
Is there any other common option for this parameter? If no, we could consider remove it.

Reply:
The ogb.smile2graph is commonly used, but because the smile2graph is used to init input node/edge feature, maybe there is other optimal way to init, I think we should remain this parameter   

> Graph: We provide molecules as the [SMILES strings](https://en.wikipedia.org/wiki/Simplified_molecular-input_line-entry_system), from which 2D molecule graphs (nodes are atoms and edges are chemical bonds) as well as molecular fingerprints (hand-engineered molecular feature developed by the chemistry community) can be obtained. The conversion requires [rdkit Python package](https://www.rdkit.org/docs/Install.html) to be installed. By default, we follow OGB to convert the SMILES string into a molecular graph representation (see code [here](https://github.com/snap-stanford/ogb/blob/master/ogb/utils/mol.py#L6)), where each node is associated with a 9-dimensional feature (e.g., atomic number, chirality) and each edge comes with a 3-dimensional feature (e.g., bond type, bond stereochemistry), although the optimal set of input graph features remains to be explored.